tests/baselines/reference/noImplicitAnyIndexing.types

```diff
@@ -39,15 +39,15 @@ var strRepresentation4 = MyEmusEnum["emu"];
 
 // Should report an implicit 'any'.
 var x = {}["hi"];
->x : any
->{}["hi"] : any
+>x : undefined
+>{}["hi"] : undefined

```

Comment:
`var x = {}["hi"]` will be `undefined` by this PR's change.

(I will delete the test cases which is for `any` type if it's appropriate)